include/CISolver/CISolver.hpp

```diff
@@ -0,0 +1,52 @@
+#ifndef GQCG_CISOLVER_HPP
+#define GQCG_CISOLVER_HPP
+
+
+#include <HamiltonianBuilder/HamiltonianBuilder.hpp>
+#include <HamiltonianParameters/HamiltonianParameters.hpp>
+#include <WaveFunction.hpp>
+#include <numopt.hpp>
+
+namespace GQCG {
+
+
+/**
+ *  Class which requires a HamiltonianBuilder and HamiltonianParameters
+ *  so that it can find a set of eigenvalues and -vectors for the Hamiltonian (which cannot always be stored in memory)
+ */
+class CISolver {
+private:
+    HamiltonianBuilder* hamiltonian_builder;
+    HamiltonianParameters hamiltonian_parameters;
+
+    std::vector<numopt::eigenproblem::Eigenpair> eigenpairs;  // eigenvalues and -vectors
+
+public:
+    // CONSTRUCTOR
+    /**
+     *  Constructor given a @param hamiltonian_builder and @param hamiltonian_parameters
+     */
+    CISolver(HamiltonianBuilder& hamiltonian_builder, HamiltonianParameters& hamiltonian_parameters);
+
+
+    // GETTERS
+    std::vector<numopt::eigenproblem::Eigenpair> get_eigenpairs() { return this->eigenpairs; }
+    numopt::eigenproblem::Eigenpair get_eigenpair(size_t index = 0) { return this->eigenpairs[index]; }
+
+
+    // PUBLIC METHODS
+    /**
+     *  solves the CI problem, setting the eigenpairs
+     */
+    void solve(numopt::eigenproblem::BaseSolverOptions& solver_options);
```

Comment:
const reference

Reply:
currently fuction get_solver_type() is not marked const